src/check_run.c

```diff
@@ -158,8 +159,47 @@ static void srunner_run_end(SRunner * sr,
     set_fork_status(CK_FORK);
 }
 
+/*
+ * Helper func to compare two lists of tags and return true if there is a
+ * common tag.
+ */
+static unsigned int matching_tag(const char *tags1_orig, const char *tags2_orig)
```

Comment:
Why not handle the splitting of tags from a string into a list in tcase_add_tags()? That may make this easier.